src/lib/containers/UserOrTeamBadge.tsx

```diff
@@ -34,31 +33,26 @@ export default function UserOrTeamBadge(props: UserOrTeamBadgeProps) {
     UserGroupHeader | undefined
   >(providedUserGroupHeader)
 
+  const { data: fetchedUserGroupHeader } = useGetUserGroupHeader(
+    (principalId ?? '').toString(),
+    {
+      enabled: !providedUserGroupHeader,
+    },
+  )
```

Comment:
We were rendering many UserOrTeamBadges with the same ID, resulting in many duplicate requests. Use react-query to cache and prevent duplicate requests.

Reply:
Nice!